### PR TITLE
Tests refactoring

### LINKED ISSUE #17: Tests refactoring
- Create a common way to mock modules
- Extract common statements to test utils

### PR DESCRIPTION
**Pull request by gohabereg**
Resolves #17

### CODE REVIEW

File: src/entities/BlockNode/BlockNode.spec.ts [hunk 1]
```diff
@@ -1,34 +1,69 @@
...
-import { TextInlineNode } from '../inline-fragments/TextInlineNode';
+import { BlockTune, BlockTuneName } from '../BlockTune';
 import { ValueNode } from '../ValueNode';
 
 import type { EditorDocument } from '../EditorDocument';
 import type { BlockTuneConstructorParameters } from '../BlockTune/types';
-import type { TextInlineNodeConstructorParameters } from '../inline-fragments/TextInlineNode';
 import type { ValueNodeConstructorParameters } from '../ValueNode';
+import { RootInlineNode } from '../inline-fragments';
+import { describe } from '@jest/globals';
```
Comment (ilyamore88) [inline src/entities/BlockNode/BlockNode.spec.ts:10]:
```suggestion
```

File: src/entities/BlockNode/BlockNode.spec.ts [hunk 2]
```diff
@@ -1,34 +1,69 @@
...
+jest.mock('../inline-fragments', () => ({
+  RootInlineNode: jest.fn().mockImplementation(() => ({
+    get serialized() {
+      return {};
+    },
+  }) as RootInlineNode),
+}));
+
+jest.mock('../ValueNode', () => ({
+  ValueNode: (class {
```
Comment (ilyamore88) [inline src/entities/BlockNode/BlockNode.spec.ts:32]: Why don't you use jest.fn() here?

File: src/entities/BlockTune/BlockTune.spec.ts [hunk 3]
```diff
@@ -1,11 +1,22 @@
-import { createBlockTuneName } from './index';
-import { createBlockTuneMock } from '../../utils/mocks/createBlockTuneMock';
+import { BlockTune, createBlockTuneName } from './index';
+import { describe } from '@jest/globals';
```
Comment (ilyamore88) [inline src/entities/BlockTune/BlockTune.spec.ts:2]:
```suggestion
```

File: src/entities/inline-fragments/mixins/ParentNode/index.ts [hunk 4]
```diff
@@ -121,6 +121,10 @@ export function ParentNode<C extends { new(...args: any[]): InlineNode }>(constr
     public removeChild(child: ChildNode): void {
       const index = this.children.indexOf(child);
 
+      if (index === -1) {
+        return;
```
Comment (ilyamore88) [inline src/entities/inline-fragments/mixins/ParentNode/index.ts:125]: Maybe throw error here?

File: src/entities/inline-fragments/specs/ParentInlineNode.spec.ts [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+      expect(result).toHaveProperty('text');
+      expect(result).toHaveProperty('fragments');
+    });
+  });
+
+  describe('.getText()', () => {
+    const start = 3;
+    const end = 5;
+
+    it('should throw a error if start index is invalid', () => {
```
Comment (ilyamore88) [inline src/entities/inline-fragments/specs/ParentInlineNode.spec.ts:69]:
```suggestion
    it('should throw an error if start index is invalid', () => {
```

File: src/entities/BlockNode/BlockNode.spec.ts (hunk 2 above)
Comment (gohabereg) [reply src/entities/BlockNode/BlockNode.spec.ts:32]: It's better to mock classes with classes I believe, I'll also move this to a mock file

File: src/entities/inline-fragments/mixins/ParentNode/index.ts (hunk 4 above)
Comment (gohabereg) [reply src/entities/inline-fragments/mixins/ParentNode/index.ts:125]: No, there is a recursion to support transparent tree operations, so this is an exit condition

File: src/entities/BlockNode/BlockNode.spec.ts (hunk 2 above)
Comment (ilyamore88) [reply src/entities/BlockNode/BlockNode.spec.ts]: Looks much better now

**Review by ilyamore88 — APPROVED** [review]

File: src/entities/BlockNode/__mocks__/index.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Mock for BlockNode class
+ */
+export class BlockNode {
```
Comment (neSpecc) [inline src/entities/BlockNode/__mocks__/index.ts:4]: Maybe BlockNodeMock?

Comment (gohabereg) [reply src/entities/BlockNode/__mocks__/index.ts:4]: Export should match the original file

**Review by neSpecc — APPROVED** [review]

**Pull request merged**